tests/trainer/test_trainer.py

```diff
@@ -2496,6 +2496,9 @@ def test_accelerator_config_empty(self):
             self.assertEqual(trainer.accelerator.even_batches, True)
             self.assertEqual(trainer.accelerator.use_seedable_sampler, True)
 
+            # gradient accumulation kwargs wil configure the gradient_state
+            self.assertNotIn("sync_each_batch", trainer.accelerator.gradient_state.plugin_kwargs)
```

Comment:
We need a test for the accelerate version here, 0.28.0

Reply:
got it.. my plan is to put a flag like so in `TrainerIntegrationTest`

```python
class TrainerIntegrationTest(TestCasePlus, TrainerIntegrationCommon):

    FLAG_ACCELERATOR_SUPPORT_GRAD_ACCUM_KWARGS = (
        version.parse(importlib.metadata.version("accelerate")) >= version.parse("0.28")
    )
```

Then we can guard the test like so:

```python
def test_accelerator_config_empty(self):
    # Checks that a config can be made with the defaults if not passed
    with tempfile.TemporaryDirectory() as tmp_dir:
        config = RegressionModelConfig(a=1.5, b=2.5)
        model = RegressionPreTrainedModel(config)
        eval_dataset = SampleIterableDataset()

        # Leaves one option as something *not* basic
        args = RegressionTrainingArguments(
            output_dir=tmp_dir,
        )
        trainer = Trainer(model=model, args=args, eval_dataset=eval_dataset)
        # ...
        if self.FLAG_ACCELERATOR_SUPPORT_GRAD_ACCUM_KWARGS:
            # gradient accumulation kwargs configures gradient_state
            self.assertNotIn('sync_each_batch', trainer.accelerator.gradient_state.plugin_kwargs)
```

Reply:
ok will use the `is_accelerate_available` function for this.

Reply:
however because `accelerate=0.28` has not been released, in the CI we will not be testing all the guarded tests.